src/main/java/com/microsoft/azure/kusto/kafka/connect/sink/FileWriter.java

```diff
@@ -162,13 +169,16 @@ private void handleErrors(String message, Exception e) {
     }
 
     private void dumpFile() throws IOException {
-        countingStream.close();
-        currentFileDescriptor = null;
-        boolean deleted = currentFile.file.delete();
-        if (!deleted) {
-            log.warn("couldn't delete temporary file. File exists: " + currentFile.file.exists());
-        }
+        SourceFile temp = currentFile;
         currentFile = null;
+        if (temp != null) {
+            countingStream.close();
```

Comment:
This is called from rollback()
rollback() already closes `countingStream` (line 186). Isn't it bad to double close a stream?

Reply:
no